diff --git a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx
index cd6ea6b..3186b68 100644
--- a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx
+++ b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx
@@ -235,15 +235,25 @@ const BottomSheetComponent = forwardRef<BottomSheet, BottomSheetProps>(
 
       return closedPosition;
     }, [$modal, detached, bottomInset]);
+
+    const animatedPosition = useSharedValue(INITIAL_POSITION);
+    const animatedNextPosition = useSharedValue(INITIAL_VALUE);
+    const animatedNextPositionIndex = useSharedValue(0);
+    
     const animatedSheetHeight = useDerivedValue(
-      () => animatedContainerHeight.value - animatedHighestSnapPoint.value
+      () => {
+        const snapPoint = animatedSnapPoints.value.find(
+          (v) => animatedPosition.value === v
+        ) ?? animatedHighestSnapPoint.value;
+
+
+        return animatedContainerHeight.value - snapPoint
+      }
     );
+
     const animatedCurrentIndex = useReactiveSharedValue(
       animateOnMount ? -1 : _providedIndex
     );
-    const animatedPosition = useSharedValue(INITIAL_POSITION);
-    const animatedNextPosition = useSharedValue(INITIAL_VALUE);
-    const animatedNextPositionIndex = useSharedValue(0);
 
     // conditional
     const isAnimatedOnMount = useSharedValue(false);
