diff --git a/lib/index.js b/lib/index.js
index e207e74..eb41a50 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -1056,7 +1056,7 @@ exports.schnorr = {
 Point.BASE._setWindowSize(8);
 const crypto = {
     node: nodeCrypto,
-    web: typeof self === 'object' && 'crypto' in self ? self.crypto : undefined,
+    web: typeof self === 'object' && 'crypto' in self && 'subtle' in self['crypto'] ? self['crypto'] : undefined,
 };
 const TAGS = {
     challenge: 'BIP0340/challenge',
diff --git a/package.json b/package.json
index 422d3c8..85b4d95 100644
--- a/package.json
+++ b/package.json
@@ -27,6 +27,9 @@
   "browser": {
     "crypto": false
   },
+  "react-native": {
+    "crypto": "crypto"
+  },
   "devDependencies": {
     "@noble/hashes": "1.1.2",
     "@rollup/plugin-commonjs": "22.0.0",
