diff --git a/dist/providers/WalletConnectProvider.js b/dist/providers/WalletConnectProvider.js
index 9b5b1e2..c124b1d 100644
--- a/dist/providers/WalletConnectProvider.js
+++ b/dist/providers/WalletConnectProvider.js
@@ -20,14 +20,6 @@ export default function WalletConnectProvider({ children, renderQrcodeModal: may
         ? maybeRenderQrcodeModal
         : defaultRenderQrcodeModal), [maybeRenderQrcodeModal]);
     const open = React.useCallback(async (uri, cb) => {
-        if (Platform.OS === 'android') {
-            const canOpenURL = await Linking.canOpenURL(uri);
-            if (!canOpenURL) {
-                Linking.openURL('https://walletconnect.org/wallets');
-                throw new Error('No wallets found.');
-            }
-            await Linking.openURL(uri);
-        }
         setState({
             uri,
             visible: true,
@@ -220,5 +212,5 @@ export default function WalletConnectProvider({ children, renderQrcodeModal: may
     ]);
     return (React.createElement(WalletConnectContext.Provider, { value: value },
         !!children && children,
-        Platform.OS !== 'android' && renderQrcodeModal(modalProps)));
+        renderQrcodeModal(modalProps)));
 }
