diff --git a/dist/index.js b/dist/index.js
index b9015a6..8d37a73 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -100,8 +100,10 @@ var Lottie = function (_React$Component) {
     value: function componentWillUpdate(nextProps /* , nextState */) {
       /* Recreate the animation handle if the data is changed */
       if (this.options.animationData !== nextProps.options.animationData) {
-        this.deRegisterEvents(this.props.eventListeners);
-        this.destroy();
+        if (this.anim !== null) {
+          this.deRegisterEvents(this.props.eventListeners);
+          this.destroy();
+        }
         this.options = (0, _extends3.default)({}, this.options, nextProps.options);
         this.anim = _lottieWeb2.default.loadAnimation(this.options);
         this.registerEvents(nextProps.eventListeners);
