diff --git a/node_modules/react-native-fast-crypto/android/src/main/cpp/crypto_bridge.cpp b/node_modules/react-native-fast-crypto/android/src/main/cpp/crypto_bridge.cpp
index ddafe67..b898443 100644
--- a/node_modules/react-native-fast-crypto/android/src/main/cpp/crypto_bridge.cpp
+++ b/node_modules/react-native-fast-crypto/android/src/main/cpp/crypto_bridge.cpp
@@ -122,8 +122,8 @@ static const unsigned char pr2six[256] =
 int Base64decode_len(const char *bufcoded)
 {
     int nbytesdecoded;
-    register const unsigned char *bufin;
-    register int nprbytes;
+    const unsigned char *bufin;
+    int nprbytes;
 
     bufin = (const unsigned char *) bufcoded;
     while (pr2six[*(bufin++)] <= 63);
@@ -137,9 +137,9 @@ int Base64decode_len(const char *bufcoded)
 int Base64decode(char *bufplain, const char *bufcoded)
 {
     int nbytesdecoded;
-    register const unsigned char *bufin;
-    register unsigned char *bufout;
-    register int nprbytes;
+    const unsigned char *bufin;
+    unsigned char *bufout;
+    int nprbytes;
 
     bufin = (const unsigned char *) bufcoded;
     while (pr2six[*(bufin++)] <= 63);
