diff --git a/ios/RNSScreenStack.mm b/ios/RNSScreenStack.mm
index 6c3afe6..f0e2568 100644
--- a/ios/RNSScreenStack.mm
+++ b/ios/RNSScreenStack.mm
@@ -707,9 +707,15 @@ namespace react = facebook::react;
   // RNSPanGestureRecognizer will receive events iff topScreen.fullScreenSwipeEnabled == YES;
   // Events are filtered in gestureRecognizer:shouldReceivePressOrTouchEvent: method
   if ([gestureRecognizer isKindOfClass:[RNSPanGestureRecognizer class]]) {
+    RNSPanGestureRecognizer *panRecognizer = gestureRecognizer;
     if ([self isInGestureResponseDistance:gestureRecognizer topScreen:topScreen]) {
       _isFullWidthSwiping = YES;
       [self cancelTouchesInParent];
+      CGPoint translation = [panRecognizer translationInView:_controller.view];
+      // Ignore swiping "forward" (RTL)
+      if (translation.x < 0) {
+        return NO;
+      }
       return YES;
     }
     return NO;
@@ -763,7 +769,7 @@ namespace react = facebook::react;
   RNSPanGestureRecognizer *panRecognizer = [[RNSPanGestureRecognizer alloc] initWithTarget:self
                                                                                     action:@selector(handleSwipe:)];
   panRecognizer.delegate = self;
-  [self addGestureRecognizer:panRecognizer];
+  [_controller.view addGestureRecognizer:panRecognizer];
 }
 
 - (void)handleSwipe:(UIPanGestureRecognizer *)gestureRecognizer
