diff --git a/node_modules/ts-node-dev/lib/index.js b/node_modules/ts-node-dev/lib/index.js
index c90b187..a621bf6 100644
--- a/node_modules/ts-node-dev/lib/index.js
+++ b/node_modules/ts-node-dev/lib/index.js
@@ -194,7 +194,10 @@ exports.runDev = function (script, scriptArgs, nodeArgs, opts) {
             kill(child.pid);
         }
         else {
-            child.kill('SIGTERM');
+            // Using SIGKILL because content node doesn't stop in some cases when using SIGTERM
+            // Specifically when libs is linked and triggers a restart while audius-compose
+            // is starting up
+            child.kill('SIGKILL');
         }
     };
     function stop(willTerminate) {
