diff --git a/node_modules/react-native-track-player/android/.build.gradle.swp b/node_modules/react-native-track-player/android/.build.gradle.swp
new file mode 100644
index 0000000..6b9f9c6
Binary files /dev/null and b/node_modules/react-native-track-player/android/.build.gradle.swp differ
diff --git a/node_modules/react-native-track-player/android/build.gradle b/node_modules/react-native-track-player/android/build.gradle
index ad99343..a85a2ff 100644
--- a/node_modules/react-native-track-player/android/build.gradle
+++ b/node_modules/react-native-track-player/android/build.gradle
@@ -49,7 +49,7 @@ repositories {
 }
 
 dependencies {
-    implementation 'com.github.DoubleSymmetry:KotlinAudio:v1.2.4'
+    implementation 'com.github.DoubleSymmetry:KotlinAudio:dd4bdab147c90fffcff7800790dbfcb41291d29b'
 //    implementation "com.github.doublesymmetry:kotlin-audio:1.2.2"
 
     //noinspection GradleDynamicVersion
