diff --git a/node_modules/react-native-screens/ios/RNSScreenStack.m b/node_modules/react-native-screens/ios/RNSScreenStack.m
index 47c8f8d..e63bc95 100644
--- a/node_modules/react-native-screens/ios/RNSScreenStack.m
+++ b/node_modules/react-native-screens/ios/RNSScreenStack.m
@@ -602,8 +602,14 @@ - (BOOL)gestureRecognizerShouldBegin:(UIGestureRecognizer *)gestureRecognizer
     // we want only `RNSPanGestureRecognizer` to be able to recognize when
     // `fullScreenSwipeEnabled` is set
     if ([gestureRecognizer isKindOfClass:[RNSPanGestureRecognizer class]]) {
+      RNSPanGestureRecognizer *panRecognizer = gestureRecognizer;
       _isFullWidthSwiping = YES;
       [self cancelTouchesInParent];
+      CGPoint translation = [panRecognizer translationInView:_controller.view];
+      // Ignore swiping "forward" (RTL)
+      if (translation.x < 0) {
+        return NO;
+      }
       return YES;
     }
     return NO;
@@ -657,7 +663,7 @@ - (void)setupGestureHandlers
   RNSPanGestureRecognizer *panRecognizer = [[RNSPanGestureRecognizer alloc] initWithTarget:self
                                                                                     action:@selector(handleSwipe:)];
   panRecognizer.delegate = self;
-  [self addGestureRecognizer:panRecognizer];
+  [_controller.view addGestureRecognizer:panRecognizer];
 }
 
 - (void)handleSwipe:(UIPanGestureRecognizer *)gestureRecognizer
