diff --git a/node_modules/level-blobs/node_modules/readable-stream/lib/_stream_readable.js b/node_modules/level-blobs/node_modules/readable-stream/lib/_stream_readable.js
index 19ab358..28be3eb 100644
--- a/node_modules/level-blobs/node_modules/readable-stream/lib/_stream_readable.js
+++ b/node_modules/level-blobs/node_modules/readable-stream/lib/_stream_readable.js
@@ -40,7 +40,7 @@ if (!EE.listenerCount) EE.listenerCount = function(emitter, type) {
 };
 /*</replacement>*/
 
-var Stream = require('stream');
+var Stream = require('node:stream');
 
 /*<replacement>*/
 var util = require('core-util-is');
diff --git a/node_modules/level-blobs/node_modules/readable-stream/lib/_stream_writable.js b/node_modules/level-blobs/node_modules/readable-stream/lib/_stream_writable.js
index db8539c..317779c 100644
--- a/node_modules/level-blobs/node_modules/readable-stream/lib/_stream_writable.js
+++ b/node_modules/level-blobs/node_modules/readable-stream/lib/_stream_writable.js
@@ -37,7 +37,7 @@ var util = require('core-util-is');
 util.inherits = require('inherits');
 /*</replacement>*/
 
-var Stream = require('stream');
+var Stream = require('node:stream');
 
 util.inherits(Writable, Stream);
 
diff --git a/node_modules/level-blobs/node_modules/readable-stream/readable.js b/node_modules/level-blobs/node_modules/readable-stream/readable.js
index 2a8b5c6..be97000 100644
--- a/node_modules/level-blobs/node_modules/readable-stream/readable.js
+++ b/node_modules/level-blobs/node_modules/readable-stream/readable.js
@@ -1,10 +1,10 @@
 exports = module.exports = require('./lib/_stream_readable.js');
-exports.Stream = require('stream');
+exports.Stream = require('node:stream');
 exports.Readable = exports;
 exports.Writable = require('./lib/_stream_writable.js');
 exports.Duplex = require('./lib/_stream_duplex.js');
 exports.Transform = require('./lib/_stream_transform.js');
 exports.PassThrough = require('./lib/_stream_passthrough.js');
 if (!process.browser && process.env.READABLE_STREAM === 'disable') {
-  module.exports = require('stream');
+  module.exports = require('node:stream');
 }
