diff --git a/node_modules/@react-navigation/drawer/src/views/legacy/Drawer.tsx b/node_modules/@react-navigation/drawer/src/views/legacy/Drawer.tsx
index dd4ae41..f3d86af 100644
--- a/node_modules/@react-navigation/drawer/src/views/legacy/Drawer.tsx
+++ b/node_modules/@react-navigation/drawer/src/views/legacy/Drawer.tsx
@@ -528,11 +528,30 @@ export default class DrawerView extends React.Component<DrawerProps> {
 
     const progress = drawerType === 'permanent' ? ANIMATED_ONE : this.progress;
 
+    const failOffsetXWhenClosed = swipeEnabled
+      ? [-1, Number.MAX_SAFE_INTEGER]
+      : [-1, 1]
+
     return (
       <DrawerProgressContext.Provider value={progress}>
         <PanGestureHandler
-          activeOffsetX={[-SWIPE_DISTANCE_MINIMUM, SWIPE_DISTANCE_MINIMUM]}
-          failOffsetY={[-SWIPE_DISTANCE_MINIMUM, SWIPE_DISTANCE_MINIMUM]}
+          // When the drawer is closed, allow swipe open.
+          // When the drawer is open, allow swipe closed.
+          activeOffsetX={
+            isOpen
+              ? [-SWIPE_DISTANCE_MINIMUM, Number.MAX_SAFE_INTEGER]
+              : [-1 * Number.MAX_SAFE_INTEGER, SWIPE_DISTANCE_MINIMUM]
+          }
+          failOffsetX={
+            isOpen
+              ? [-1 * Number.MAX_SAFE_INTEGER, Number.MAX_SAFE_INTEGER]
+              : failOffsetXWhenClosed
+          }
+          failOffsetY={
+            isOpen 
+              ? [-1 * Number.MAX_SAFE_INTEGER, Number.MAX_SAFE_INTEGER] 
+              : [-SWIPE_DISTANCE_MINIMUM, SWIPE_DISTANCE_MINIMUM]
+          }
           onGestureEvent={this.handleGestureEvent}
           onHandlerStateChange={this.handleGestureStateChange}
           hitSlop={hitSlop}
