diff --git a/node_modules/react-native-gesture-handler/ios/RNGestureHandler.h b/node_modules/react-native-gesture-handler/ios/RNGestureHandler.h
index 698f0d0..77ec682 100644
--- a/node_modules/react-native-gesture-handler/ios/RNGestureHandler.h
+++ b/node_modules/react-native-gesture-handler/ios/RNGestureHandler.h
@@ -68,6 +68,7 @@
 @property (nonatomic) BOOL shouldCancelWhenOutside;
 @property (nonatomic) BOOL needsPointerData;
 @property (nonatomic) BOOL manualActivation;
+@property (nonatomic) BOOL frozen;
 
 - (void)bindToView:(nonnull UIView *)view;
 - (void)unbindFromView;
@@ -87,4 +88,7 @@
 - (void)sendEvent:(nonnull RNGestureHandlerStateChange *)event;
 - (void)sendTouchEventInState:(RNGestureHandlerState)state forViewWithTag:(nonnull NSNumber *)reactTag;
 
++ (NSHashTable<RNGestureHandler *> *)getGestureHandlers;
+
+
 @end
diff --git a/node_modules/react-native-gesture-handler/ios/RNGestureHandler.m b/node_modules/react-native-gesture-handler/ios/RNGestureHandler.m
index 39a6e42..77f23fd 100644
--- a/node_modules/react-native-gesture-handler/ios/RNGestureHandler.m
+++ b/node_modules/react-native-gesture-handler/ios/RNGestureHandler.m
@@ -81,6 +81,7 @@ - (instancetype)initWithTag:(NSNumber *)tag
     _hitSlop = RNGHHitSlopEmpty;
     _state = RNGestureHandlerStateBegan;
     _manualActivationRecognizer = nil;
+    _frozen = NO;
 
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
@@ -167,8 +168,10 @@ - (void)configure:(NSDictionary *)config
 
 - (void)setEnabled:(BOOL)enabled
 {
-  _enabled = enabled;
-  self.recognizer.enabled = enabled;
+  if (_frozen == NO) {
+    _enabled = enabled;
+    self.recognizer.enabled = enabled;
+  }
 }
 
 - (void)bindToView:(UIView *)view
@@ -490,4 +493,9 @@ - (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldReceive
   return YES;
 }
 
++ (NSHashTable<RNGestureHandler *> *)getGestureHandlers
+{
+  return allGestureHandlers;
+}
+
 @end
