diff --git a/node_modules/react-native-radial-gradient/index.js b/node_modules/react-native-radial-gradient/index.js
index 4be67fa..ea543dd 100644
--- a/node_modules/react-native-radial-gradient/index.js
+++ b/node_modules/react-native-radial-gradient/index.js
@@ -5,7 +5,13 @@ import {ViewPropTypes} from 'deprecated-react-native-prop-types';
 
 export default class RadialGradient extends Component {
     static propTypes = {
-        center: PropTypes.arrayOf(PropTypes.number),
+        center: PropTypes.oneOfType([
+            PropTypes.arrayOf(PropTypes.number),
+            PropTypes.shape({
+                x: PropTypes.number,
+                y: PropTypes.number
+            })
+        ]),
         colors: PropTypes.arrayOf(PropTypes.string),
         stops: PropTypes.arrayOf(PropTypes.number),
         radius: PropTypes.number,
