diff --git a/node_modules/react-native-gesture-handler/ios/RNGestureHandler.h b/node_modules/react-native-gesture-handler/ios/RNGestureHandler.h
index f952c15..2f4e9bd 100644
--- a/node_modules/react-native-gesture-handler/ios/RNGestureHandler.h
+++ b/node_modules/react-native-gesture-handler/ios/RNGestureHandler.h
@@ -55,6 +55,7 @@ if (value != nil) recognizer.prop = [RCTConvert type:value]; \
 @property (nonatomic, readonly, nullable) UIGestureRecognizer *recognizer;
 @property (nonatomic) BOOL enabled;
 @property(nonatomic) BOOL shouldCancelWhenOutside;
+@property (nonatomic) BOOL frozen;
 
 - (void)bindToView:(nonnull UIView *)view;
 - (void)unbindFromView;
@@ -69,5 +70,7 @@ if (value != nil) recognizer.prop = [RCTConvert type:value]; \
            forViewWithTag:(nonnull NSNumber *)reactTag
             withExtraData:(RNGestureHandlerEventExtraData *)extraData;
 
++ (NSHashTable<RNGestureHandler *> *)getGestureHandlers;
+
 @end
 
diff --git a/node_modules/react-native-gesture-handler/ios/RNGestureHandler.m b/node_modules/react-native-gesture-handler/ios/RNGestureHandler.m
index 998c059..b239ba7 100644
--- a/node_modules/react-native-gesture-handler/ios/RNGestureHandler.m
+++ b/node_modules/react-native-gesture-handler/ios/RNGestureHandler.m
@@ -73,6 +73,7 @@ - (instancetype)initWithTag:(NSNumber *)tag
         _tag = tag;
         _lastState = RNGestureHandlerStateUndetermined;
         _hitSlop = RNGHHitSlopEmpty;
+        _frozen = NO;
 
         static dispatch_once_t onceToken;
         dispatch_once(&onceToken, ^{
@@ -132,8 +133,10 @@ - (void)configure:(NSDictionary *)config
 
 - (void)setEnabled:(BOOL)enabled
 {
-    _enabled = enabled;
-    self.recognizer.enabled = enabled;
+    if (_frozen == NO) {
+        _enabled = enabled;
+        self.recognizer.enabled = enabled;
+    }
 }
 
 - (void)bindToView:(UIView *)view
@@ -341,4 +344,9 @@ - (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldReceive
     return YES;
 }
 
++ (NSHashTable<RNGestureHandler *> *)getGestureHandlers
+{
+    return allGestureHandlers;
+}
+
 @end
