diff --git a/node_modules/react-native-google-cast/react-native-google-cast.podspec b/node_modules/react-native-google-cast/react-native-google-cast.podspec
index d1948aa..39212a4 100644
--- a/node_modules/react-native-google-cast/react-native-google-cast.podspec
+++ b/node_modules/react-native-google-cast/react-native-google-cast.podspec
@@ -32,7 +32,7 @@ Pod::Spec.new do |s|
 
   s.subspec 'NoBluetoothArm' do |ss|
     ss.dependency "#{package['name']}/RNGoogleCast"
-    ss.dependency 'google-cast-sdk-dynamic-xcframework-no-bluetooth'
+    ss.dependency 'google-cast-sdk-dynamic-xcframework-no-bluetooth', '4.7.1'
   end
 
   s.subspec 'RNGoogleCast' do |ss|
