diff --git a/node_modules/react-native/Libraries/Animated/AnimatedEvent.js b/node_modules/react-native/Libraries/Animated/AnimatedEvent.js
index 7279c2a..d798b42 100644
--- a/node_modules/react-native/Libraries/Animated/AnimatedEvent.js
+++ b/node_modules/react-native/Libraries/Animated/AnimatedEvent.js
@@ -251,6 +251,6 @@ export class AnimatedEvent {
   }
 
   _callListeners = (...args: any) => {
-    this._listeners.forEach(listener => listener(...args));
+    this._listeners.forEach(listener => listener?(...args));
   };
 }
