diff --git a/node_modules/@openzeppelin/upgrades/contracts/Initializable.sol b/node_modules/@openzeppelin/upgrades/contracts/Initializable.sol
index 09a4a05..445a2e4 100644
--- a/node_modules/@openzeppelin/upgrades/contracts/Initializable.sol
+++ b/node_modules/@openzeppelin/upgrades/contracts/Initializable.sol
@@ -13,7 +13,11 @@ pragma solidity >=0.4.24 <0.7.0;
  * a parent initializer twice, or ensure that all initializers are idempotent,
  * because this is not dealt with automatically as with constructors.
  */
-contract Initializable {
+ contract Initializable {
+  address private proxyAdmin;
+    
+  uint256 private filler1;
+  uint256 private filler2;
 
   /**
    * @dev Indicates that the contract has been initialized.
@@ -29,6 +33,7 @@ contract Initializable {
    * @dev Modifier to use in the initializer function of a contract.
    */
   modifier initializer() {
+    require(msg.sender == proxyAdmin, "Only proxy admin can initialize");
     require(initializing || isConstructor() || !initialized, "Contract instance has already been initialized");
 
     bool isTopLevelCall = !initializing;
@@ -58,5 +63,5 @@ contract Initializable {
   }
 
   // Reserved storage space to allow for layout changes in the future.
-  uint256[50] private ______gap;
+  uint256[47] private ______gap;
 }
\ No newline at end of file
